source/dub/dependency.d

```diff
@@ -258,7 +276,12 @@ struct Dependency {
 	*/
 	string toString()()
 	const {
-		auto ret = versionSpec;
+		string ret;
+
+		if (!repository.empty) {
+			ret ~= "#";
+		}
+		ret ~= versionSpec;
```

Comment:
Not an expert on all the options here, but this formulation looks questionable: is it legit for the git version spec to be followed by extra stuff (which will happen if `optional` is true, or if `!path.empty`)?

Also, why the `#` symbol?